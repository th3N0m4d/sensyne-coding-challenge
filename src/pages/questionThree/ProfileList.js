import { makeStyles } from '@material-ui/core/styles'
import List from '@material-ui/core/List'
import Paper from '@material-ui/core/Paper'
import React from 'react'
import PropTypes from 'prop-types'

import Profile from './Profile'

const useStyles = makeStyles({
  container: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%'
  },
  root: {
    width: '100%'
  },
  inline: {
    display: 'inline'
  }
})

const propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      Icon: PropTypes.any,
      name: PropTypes.string,
      species: PropTypes.string,
      divider: PropTypes.bool,
      description: PropTypes.string,
      id: PropTypes.any
    })
  )
}

const defaultProps = {
  items: []
}

const ProfileList = ({ items }) => {
  const classes = useStyles()

  return (
    <div className={classes.container}>
      <Paper>
        <List className={classes.root}>
          {items.map((item, i) => (
            <Profile {...item} key={item.id} divider={i !== items.length - 1} />
		      ))}
        </List>
      </Paper>
    </div>
  )
}

ProfileList.propTypes = propTypes
ProfileList.defaultProps = defaultProps

export default ProfileList
