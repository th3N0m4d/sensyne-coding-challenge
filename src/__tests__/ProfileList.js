import React from 'react'
import renderer from 'react-test-renderer'
import ProfileList from '../pages/questionThree/ProfileList'

const expectToMatchSnapshot = component => {
  expect(renderer.create(component).toJSON()).toMatchSnapshot()
}

describe('ProfileList', () => {
  it('should render component', () => {
    expectToMatchSnapshot(<ProfileList />)
  })

  it('should render two items', () => {
    const items = [
      { id: '1', name: 'Sponge Bob' },
      { id: '2', name: 'Patryk Star' }
    ]
    expectToMatchSnapshot(<ProfileList items={items} />)
  })
})
