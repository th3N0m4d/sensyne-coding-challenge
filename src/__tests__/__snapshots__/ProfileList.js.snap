// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileList should render component 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      className="MuiList-root makeStyles-root-2 MuiList-padding"
    />
  </div>
</div>
`;

exports[`ProfileList should render two items 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      className="MuiList-root makeStyles-root-2 MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            className="MuiAvatar-root MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Sponge Bob: Other
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            <div>
              <h6
                className="MuiTypography-root makeStyles-header-35 MuiTypography-subtitle2"
              >
                Description:
              </h6>
              <p
                className="MuiTypography-root makeStyles-label-36 MuiTypography-body2"
              />
            </div>
            <div>
              <h6
                className="MuiTypography-root makeStyles-header-35 MuiTypography-subtitle2"
              >
                Guid:
              </h6>
              <p
                className="MuiTypography-root makeStyles-label-36 MuiTypography-body2"
              >
                1
              </p>
            </div>
          </p>
        </div>
      </li>
      <hr
        className="MuiDivider-root MuiDivider-middle"
      />
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            className="MuiAvatar-root MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Patryk Star: Other
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            <div>
              <h6
                className="MuiTypography-root makeStyles-header-35 MuiTypography-subtitle2"
              >
                Description:
              </h6>
              <p
                className="MuiTypography-root makeStyles-label-36 MuiTypography-body2"
              />
            </div>
            <div>
              <h6
                className="MuiTypography-root makeStyles-header-35 MuiTypography-subtitle2"
              >
                Guid:
              </h6>
              <p
                className="MuiTypography-root makeStyles-label-36 MuiTypography-body2"
              >
                2
              </p>
            </div>
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
